x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/ExecutionManager.java

```diff
@@ -214,6 +224,23 @@ public Executable assemble(List<List<Attribute>> listOfKeys, List<PhysicalPlan>
         return new SampleIterator(new PITAwareQueryClient(session), criteria, cfg.fetchSize(), limit, session.circuitBreaker());
     }
 
+    private int indexOfFirstPositive(boolean[] missing) {
+        for (int i = 0; i < missing.length; i++) {
+            if (missing[i] == false) {
+                return i;
+            }
+        }
+        throw new IllegalStateException("Invalid sequence query: at least one event has to be positive");
+    }
```

Comment:
I think this entire check needs to happen at parsing time, likely in the LogicalPlanBuilder.

Reply:
I agree with you about throwing the exception at parsing time, I'm adding it to LogicalPlanBuilder.
I don't like very much the idea of calculating the index of the first missing event at parsing time and passing it to the constructor, mostly because it would allow to pass conflicting information (ie. a `boolean[] missing` and the wrong `indexOfFirstPositive`), so I'd prefer to keep the calculation here, maybe doing it only once at the beginning.